### PR TITLE
Documentation page with ipyleaflet example

### PR DESCRIPTION
**Pull request by sandertyu**
This is basically a copy paste of Jason's #262 for pythreejs, except I've changed relevant portions to instead be about ipyleaflet. The `index.rst` file might end up conflicting but I'm not 100% sure.

### CODE REVIEW

**Comment by moorepants** [comment]
Does this one fully work?

**Comment by sandertyu** [comment]
> Does this one fully work?

Yes this ipyleaflet example works 100%, and I don't know of any issues with using ipyleaflet in Thebe.

File: docs/ipyleaflet_example.rst [hunk 1]
```diff
@@ -0,0 +1,127 @@
+=================
+Ipyleaflet Example
+=================
+
+Thebe can display output from ipyleaflet_ Jupyter widgets. This example_ is repurposed from the
+ipyleaflet documentation and is licensed under the BSD 3 Clause.
```
Comment (moorepants) [inline docs/ipyleaflet_example.rst:6]: ipyleaflet is MIT licensed

File: docs/ipyleaflet_example.rst [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+.. _ipyleaflet: https://github.com/jupyter-widgets/ipyleaflet
+
+.. _example: https://github.com/jupyter-widgets/ipyleaflet/blob/master/examples/Heatmap.ipynb
+
+Setup
+=====
+
+Be sure to load require.min.js before any of your thebe activation code, it is required for widgets to work:
```
Comment (moorepants) [inline docs/ipyleaflet_example.rst:15]:
```suggestion
Be sure to load require.min.js before any of your thebe activation code, it is required for Jupyter widgets to work:
```

File: docs/ipyleaflet_example.rst [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+.. code-block:: html
+
+   <script type="text/x-thebe-config">
+     {
+       requestKernel: true,
+       binderOptions: {
+         repo: "jupyter-widgets/ipyleaflet",
+         ref: "0.13.3",
+         binderUrl: "https://binder.libretexts.org",
```
Comment (moorepants) [inline docs/ipyleaflet_example.rst:31]:
```suggestion
         binderUrl: "https://mybinder.org",
```

File: docs/ipyleaflet_example.rst [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+.. raw:: html
+
+   <script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
+   <script type="text/x-thebe-config">
+     {
+       requestKernel: true,
+       binderOptions: {
+         repo: "jupyter-widgets/ipyleaflet",
+         ref: "0.13.3",
+         binderUrl: "https://binder.libretexts.org",
```
Comment (moorepants) [inline docs/ipyleaflet_example.rst:70]:
```suggestion
         binderUrl: "https://mybinder.org",
```

**Comment by moorepants** [comment]
Make a `docs/examples/` directory and put this in there. I'll do the same with the others.

File: docs/ipyleaflet_example.rst (hunk 1 above)
Comment (sandertyu) [reply docs/ipyleaflet_example.rst:6]: Okay, the README and license file in their repo claim different things, but we'll go with MIT as you say and that's what the license file says.

Comment (moorepants) [reply docs/ipyleaflet_example.rst]: Oh, I might be mistaken. I just saw MIT listed from the auto detection of github. We could put whatever is in their actual license file.

Comment (davidbrochart) [reply docs/ipyleaflet_example.rst]: Yes we will update the README, but the license is MIT.

**Comment by moorepants** [comment]
LGTM

**Comment by choldgraf** [comment]
Woo thanks for hitting the green button on all of these!

**Pull request merged**